src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -823,6 +870,17 @@ public function run( $options ) {
 
 		$this->skin->after();
 
+		// Clean up the backup kept in the temp-backup directory.
+		if ( ! empty( $options['hook_extra']['temp_backup'] ) ) {
+			// Delete the backup on `shutdown` to avoid a PHP timeout.
+			add_action(
+				'shutdown',
+				function() use ( $options ) {
+					$this->delete_temp_backup( $options['hook_extra']['temp_backup'] );
+				}
+			);
```

Comment:
Change the closure to be a private method as suggested here https://github.com/WordPress/wordpress-develop/pull/2225/files#r828178375.

Reply:
Can't use private method in callback as hook is public.

Reply:
Closure removed